File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/moodgrid_provider.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
** Internal class  moodgrid_provider
*/
public class moodgrid_provider : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal moodgrid_provider(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(moodgrid_provider obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~moodgrid_provider() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_moodgrid_provider(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

  public GnMoodgridProvider get_data(uint pos) {
    GnMoodgridProvider ret = new GnMoodgridProvider(gnsdk_csharp_marshalPINVOKE.moodgrid_provider_get_data(swigCPtr, pos), true);
    return ret;
  }

  public uint count() {
    uint ret = gnsdk_csharp_marshalPINVOKE.moodgrid_provider_count(swigCPtr);
    return ret;
  }

  public moodgrid_provider() : this(gnsdk_csharp_marshalPINVOKE.new_moodgrid_provider(), true) {
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/moodgrid_provider.java

package com.gracenote.gnsdk;

/** 
** Internal class  moodgrid_provider 
*/ 
 
public class moodgrid_provider {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected moodgrid_provider(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(moodgrid_provider obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_moodgrid_provider(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public GnMoodgridProvider getData(long pos) {
    return new GnMoodgridProvider(gnsdk_javaJNI.moodgrid_provider_getData(swigCPtr, this, pos), true);
  }

  public long count() {
    return gnsdk_javaJNI.moodgrid_provider_count(swigCPtr, this);
  }

  public moodgrid_provider() {
    this(gnsdk_javaJNI.new_moodgrid_provider(), true);
  }

}
